components/BaseMultiselect.vue

```diff
@@ -0,0 +1,284 @@
+};
+
+useClickOutside(dropdown, () => {
+  isDropdownOpen.value = false;
+});
+
+const check = (sortType: string, e: Event) => {
+  const target = e.target as HTMLInputElement;
+
+  let updatedValue = [...props.modelValue];
+
+  if (target.checked) {
+    updatedValue.push(sortType);
+  } else {
+    const filteredValue = updatedValue.filter((value) => value !== sortType);
+    updatedValue = filteredValue;
+  }
+  emit("update:modelValue", updatedValue);
+};
+
+const uncheckType = (sortType: string) => {
+  let updatedValue = [...props.modelValue];
+  const filteredValue = updatedValue.filter((value) => value !== sortType);
+  updatedValue = filteredValue;
+  emit("update:modelValue", updatedValue);
+};
+
+const checkByKeyboard = (index: number) => {
+  let updatedValue = [...props.modelValue];
+
+  if (index !== -1 && !updatedValue.includes(props.fileTypes[index])) {
+    updatedValue.push(props.fileTypes[index]);
+  } else {
+    const filteredValue = updatedValue.filter(
+      (value) => value !== props.fileTypes[index]
+    );
+    updatedValue = filteredValue;
+  }
+  emit("update:modelValue", updatedValue);
+};
+</script>
+
+<template>
+  <div class="select">
+    <p class="select__label">{{ label }}</p>
+    <div ref="dropdown" class="select__dropdown">
+      <div class="select__container">
+        <div class="select__selected">
+          <template v-if="modelValue.length > 0">
+            <button
+              v-for="type in modelValue"
+              :key="type"
+              class="select__selected--type"
+              @click="uncheckType(type)"
+            >
+              {{ type }}
+              <UncheckIcon class="select__selected--icon" />
+            </button>
+          </template>
+        </div>
+        <button
+          class="select__button"
+          aria-labelledby="toggle dropdown"
+          aria-haspopup="listbox"
+          :aria-expanded="isDropdownOpen"
+          @click="toggleDropdown"
+          @keyup.esc="isDropdownOpen = false"
+          @keydown.down.prevent="highlightNext()"
+        >
+          <component :is="isDropdownOpen ? SortUp : SortDown" />
+        </button>
+      </div>
+      <ul v-if="isDropdownOpen" class="select__list" role="listbox">
+        <li
+          v-for="(type, index) in fileTypes"
+          :key="type"
+          class="select__listitem"
+          :class="{
+            'select__listitem--highlighted': highlightedIndex === index,
+          }"
+          role="option"
+          @mouseover="highlightedIndex = index"
+        >
+          <input
+            :id="type"
+            class="select__listitem--checkbox"
+            type="checkbox"
+            name="sortType"
+            :value="type"
+            :checked="modelValue.includes(type)"
+            @input="check(type, $event)"
+          />
+          <label
+            :ref="
+              (el) => {
+                if (el !== null) labelsRefs[index] = el;
+              }
+            "
```

Comment:
This :ref expression is quite long, but because of if statement I have to use {}. It was quite tricky to set ref on elements rendered from an array :)